ironfish/src/rpc/routes/chain/getBlockInfo.test.ts

```diff
@@ -0,0 +1,141 @@
+    await chain.open()
+    strategy.disableMiningReward()
+
+    const genesis = await chain.getBlock(chain.genesis)
+    Assert.isNotNull(genesis)
+
+    const blockA1 = await makeBlockAfter(chain, genesis)
+    await expect(chain).toAddBlock(blockA1)
+
+    const blockA2 = await makeBlockAfter(chain, blockA1)
+    await expect(chain).toAddBlock(blockA2)
+
+    //Get hash of blocks
+    const hash0 = genesis.header.hash.toString('hex')
+    const hash1 = blockA1.header.hash.toString('hex')
+    const hash2 = blockA2.header.hash.toString('hex')
+
+    //Find block matching hash
+    let response = await routeTest.client
+      .request<GetBlockInfoResponse>('chain/getBlockInfo', { search: hash2 })
+      .waitForEnd()
+
+    expect(response.content).toMatchObject({
+      block: {
+        hash: hash2,
+        sequence: 3,
+      },
+    })
+
+    //Now miss on a hash check
+    try {
+      await routeTest.client
+        .request<GetBlockInfoResponse>('chain/getBlockInfo', {
+          search: '123405c7492bd000d6a8312f7592737e869967c890aac22247ede00678d4a2b2',
+        })
+        .waitForEnd()
+    } catch (e: unknown) {
+      if (!(e instanceof RpcRequestError)) {
+        throw e
+      }
+      expect(e.status).toBe(400)
+      expect(e.code).toBe(ERROR_CODES.VALIDATION)
+      expect(e.message).toContain('No block found with hash')
+    }
+
+    //Find block matching sequence
+    response = await routeTest.client
+      .request<GetBlockInfoResponse>('chain/getBlockInfo', { search: '2' })
+      .waitForEnd()
+
+    expect(response.content).toMatchObject({
+      block: {
+        hash: hash1,
+        sequence: 2,
+      },
+    })
+
+    //Find block matching sequence
+    response = await routeTest.client
+      .request<GetBlockInfoResponse>('chain/getBlockInfo', { search: '-1' })
+      .waitForEnd()
+
+    expect(response.content).toMatchObject({
+      block: {
+        hash: hash2,
+        sequence: 3,
+      },
+    })
+
+    //Now miss on a sequence check
+    try {
+      await routeTest.client
+        .request<GetBlockInfoResponse>('chain/getBlockInfo', { search: '1234' })
+        .waitForEnd()
+    } catch (e: unknown) {
+      if (!(e instanceof RpcRequestError)) {
+        throw e
+      }
+      expect(e.status).toBe(400)
+      expect(e.code).toBe(ERROR_CODES.VALIDATION)
+      expect(e.message).toContain('No block found with sequence')
+    }
+
+    // Force failure of getBlock()
+    jest.spyOn(chain, 'getBlock').mockResolvedValue(null)
+
+    try {
+      await routeTest.client
+        .request<GetBlockInfoResponse>('chain/getBlockInfo', { search: hash0 })
+        .waitForEnd()
+    } catch (e: unknown) {
+      if (!(e instanceof RpcRequestError)) {
+        throw e
+      }
+      expect(e.status).toBe(400)
+      expect(e.code).toBe(ERROR_CODES.VALIDATION)
+      expect(e.message).toContain('No block with header')
+    }
+  })
+
+  it('Receives transactions from a matched block', async () => {
```

Comment:
I think you're missing an additional fixture for this test that should make it run faster. When I ran the tests locally, git showed additional changes to `getBlockInfo.test.ts.fixture`.

Reply:
Done